Migracode|Afere Precious Onome|Task Manager Exercise.

AferePreciousOnome opened the pull request:
I have made the changes you suggested.

AferePreciousOnome committed:
Done with my live session task manager exercise.

nicostuhlfauth commented on web/src/components/domains/task/TaskItem/TaskItem.jsx:
```diff
@@ -1,11 +1,68 @@
+      <button className={colorBox()}>{props.priority}</button>
+      <p className={styles.dueDate}>
+        <span>
+          {props.photos.map((photo, index) => (
```
You can directly use icons inline. As it's always the same icon we expect here, you do not need to pass it as a prop.

You can easily use icons this way:

`const MyIcon = () => (<svg>...</svg>);`

This creates a React component that returns your Icon that you can use instead of <img />. You can easily get the SVG for your icons from Figma (right-click on the Icon and select Copy/Paste as > Copy as SVG).

💡 It's recommended to use [SVGO](https://jakearchibald.github.io/svgomg/) or similar tools to minify the SVG and keep it as small as possible.

nicostuhlfauth commented on web/src/components/domains/task/TaskItem/TaskItem.jsx:
```diff
@@ -1,11 +1,68 @@
+
+      <p className={styles.assignee}>
+        <span>
+          {props.studentPhotos.map((pic, index) => (
```
Same like for the other icon...

nicostuhlfauth commented on web/src/components/domains/task/TaskItem/TaskItem.jsx:
```diff
@@ -1,11 +1,68 @@
+  dueDate: PropTypes.string.isRequired,
+  assignee: PropTypes.string.isRequired,
+  project: PropTypes.string.isRequired,
+  photos: PropTypes.arrayOf(PropTypes.string).isRequired,
+  studentPhotos: PropTypes.arrayOf(PropTypes.string).isRequired,
```
```suggestion
```

nicostuhlfauth commented on web/src/components/domains/task/TaskItem/TaskItem.jsx:
```diff
@@ -1,11 +1,68 @@
+  return (
+    <div className={styles.mainContainer}>
+      <h4>{props.ProjectName}</h4>
+      <button className={colorBox()}>{props.priority}</button>
```
I think here using a div would be the better choice. 

```suggestion
      <div className={colorBox()}>{props.priority}</div>
```

nicostuhlfauth reviewed (commented):
Hey @AferePreciousOnome 👋,

congrats on your first React PR! This looks quite good, I just left you a few comments.

General recommendations:

- do not overuse props: if we always want the same behavior, we can directly create it as part of the component. If we need more flexibility later, we can always change that. 
- you could simplify your styles, for examples creating a new class containing all shared styles of the colorBoxes, like height, width, everything not related to the color.

AferePreciousOnome committed:
i updated my code and put the icons in different component, and i also limited the use of props.

AferePreciousOnome committed:
i made changes to the button and did my final changes to my code.

AferePreciousOnome committed:
-Created a NewTask Component
-Which Include a form
-Added `.trim` to remove white spaces from input.
-Stopped submission if it contains only whitespace.
-Ensured input field reset after adding a task.
